src/components/Calculator.js

```diff
@@ -1,42 +1,75 @@
 import React, { Component } from "react";
+import calculate from "../logic/calculate";
 
 export class Calculator extends Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {calcoutput: 0,calcobj:{},calcnext:null,calctotal:null};
+  }
+  
   render() {
+    const clickbutton = (inp) => (event) => {     
+      let returnobj = calculate(this.state.calcobj,inp);
+      this.setState({
+        calcobj: returnobj
+      })
+      this.setState({
+        calcoutput: returnobj.next
+      })
+      if (returnobj.next === null && returnobj.operation === null) {
+        this.setState({
+          calcoutput: returnobj.total
+        })
+      }
+      if (returnobj.next === null && returnobj.operation !== null) {
+        this.setState({
+          calcoutput: returnobj.operation
+        })
+      }
+      if (inp === 'ac') {
+        this.setState({
+          calcobj:{},
+          calcoutput:0
+        })
+        
+      }
+  }
     return (
       <div className="calculatorcontainer">
         <div className="outputbar">
-          <span className="outputinp">0</span>
+          <span className="outputinp">{this.state.calcoutput}</span>
         </div>
 
         <div className="keys">
           <div className="row row1">
-            <div className="key key1">ac</div>
-            <div className="key key2">+/-</div>
-            <div className="key key3">%</div>
-            <div className="key key4 orange">/</div>
+            <div onClick={clickbutton('ac')} className="key key1">ac</div>
+            <div onClick={clickbutton('+/-')} className="key key2">+/-</div>
+            <div onClick={clickbutton('%')} className="key key4">%</div>
```

Comment:
[OPTIONAL] There is a bug we one tries to use the modulo operator with 0, as it is a division by 0. The original logic files have a catch for the division by 0, but not for the modulo. I think you could make it work just like that one.

Reply:
I fixed the bug, but i fixed it inside the operate.js file. I'm not sure if i should edit that file. If not, please let me know.